tests/core/asyncio/test_connect.py

```diff
@@ -5,41 +5,20 @@
 
 
 @pytest.mark.asyncio
-async def test_can_not_connect_conflicting_names():
-    own = ConnectionConfig.from_name(generate_unique_name())
-    endpoint = AsyncioEndpoint()
-    await endpoint.start_serving(own)
-
-    # We connect to our own Endpoint because for this test, it doesn't matter
-    # if we use a foreign one or our own
-    await endpoint.connect_to_endpoints(own)
-
-    # Can't connect a second time
-    with pytest.raises(ConnectionAttemptRejected):
-        await endpoint.connect_to_endpoints(own)
-
-    endpoint.stop()
+async def test_connect_to_endpoint():
+    config = ConnectionConfig.from_name(generate_unique_name())
+    async with AsyncioEndpoint.serve(config):
+        async with AsyncioEndpoint("client").run() as client:
+            await client.connect_to_endpoint(config)
+            assert client.is_connected_to(config.name)
 
 
 @pytest.mark.asyncio
 async def test_rejects_duplicates_when_connecting():
     own = ConnectionConfig.from_name(generate_unique_name())
-    endpoint = AsyncioEndpoint()
-    await endpoint.start_serving(own)
-
-    with pytest.raises(ConnectionAttemptRejected):
-        await endpoint.connect_to_endpoints(own, own)
-
-    endpoint.stop()
-
-
-@pytest.mark.asyncio
-async def test_rejects_duplicates_when_connecting_nowait():
-    own = ConnectionConfig.from_name(generate_unique_name())
-    endpoint = AsyncioEndpoint()
-    await endpoint.start_serving(own)
-
-    with pytest.raises(ConnectionAttemptRejected):
-        endpoint.connect_to_endpoints_nowait(own, own)
```

Comment:
These felt really odd to be testing using these non-core APIs so I changed to only use `connect_to_endpoint`.

I'm feeling like I want to remove the various extra `connect_to_endpointXXXXX` APIs as they feel like they add clutter and are only moderately more useful.